include/openenclave/edger8r/common.h

```diff
@@ -66,20 +66,33 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
     return result;
 }
 
-#define OE_ADD_SIZE(total, size)                                   \
+#define OE_COMPUTE_SIZE(a, b, c)                 \
+    do                                           \
+    {                                            \
+        size_t _a = (size_t)(a);                 \
+        size_t _b = (size_t)(b);                 \
+        if (_a && _b && (_a > OE_SIZE_MAX / _b)) \
+        {                                        \
+            _result = OE_INTEGER_OVERFLOW;       \
+            goto done;                           \
+        }                                        \
+        c = _a * _b;                             \
+    } while (0);
+
+#define OE_ADD_SIZE(total, argcount, argsize)                      \
     do                                                             \
     {                                                              \
-        if (sizeof(total) > sizeof(size_t) && total > OE_SIZE_MAX) \
+        size_t _addend = (size_t)(argsize);                        \
+        if (argcount > 1)                                          \
```

Comment:
Consider inverting the dependency instead of redefining the existing signature to include expanded use. It makes the semantics of the new use case clearer, and you don't need to update existing uses to a new macro signature:

```c
#define OE_ADD_ARG_SIZE(total, argcount, argsize) \
    do                                            \
    {                                             \
        size_t _total_argsize = 0;                \

        // Existing logic for OE_COMPUTE_SIZE(argcount, argsize, _total_argsize)

        OE_ADD_SIZE(total, _total_argsize);       \
     } while (0); 
```

Reply:
Fixed